TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -443,10 +445,111 @@ public static void CheckPicture(Node node, CommonDataDescriptionAndDataRedefines
             }
         }
 
+#if DOMAIN_CHECKER
+        /// <summary>
+        /// Get the top program.
+        /// </summary>
+        /// <param name="curPrg"></param>
+        /// <returns></returns>
+        static ProgramSymbol GetTopProgram(ProgramSymbol curPrg)
+        {
+            ProgramSymbol top = (ProgramSymbol)curPrg.TopParent(Symbol.Kinds.Program);
+            if (top == null || top == curPrg) return curPrg;
+            return GetTopProgram(top);
+        }
+
+        /// <summary>
+        /// Expand the top program.
+        /// </summary>
+        /// <param name="curPrg">The Current Program.</param>
+        /// <returns></returns>
+        static ProgramSymbol ExpandTopProgram(ProgramSymbol curPrg)
+        {
+            //if (ProgramSymbolTableBuilder.LastBuilder != null &&
+            //    ProgramSymbolTableBuilder.LastBuilder.Diagnostics.Count > 0)
+            //{//There was errord dont expand the program.
+            //    return curPrg;
+            //}
+
+            const ulong expandedFlag = 0x1 << 62;//I use this flag to mark an expanded program.
```

Comment:
Why 62 ?

Reply:
This will be moved in the SymbolExpander class and Symbol.Flags, ok